File: backend/src/main/java/com/kiwi/features/goals/data/UserGoalStatusDTO.java
package com.kiwi.features.goals.data;

import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode
@ToString
public class UserGoalStatusDTO {
    private Long id;
    private Long goalId;
    // Denormalized from GoalPersistence for display
    private String name;
    private String action;
    private Integer target;
    private String type;
    private String category;
    private Integer reward;
    private Integer onCompletedEntityId;
    private String onCompletedEvent;
    // Per-user status fields
    private String status;
    private String date;
    private Integer value;
}

File: mobile/app/src/main/java/com/bellako/kiwi/features/goals/data/UserGoalStatusDTO.kt
package com.bellako.kiwi.features.goals.data

data class UserGoalStatusDTO(
    val id: Long,
    val goalId: Long,
    val name: String,
    val target: Int,
    val action: String,
    val type: String,
    val category: String,
    val status: String,
    val reward: Int,
    val date: String = "",
    val value: Int = 0,
    val onCompletedEvent: String = "_",
    val onCompletedEntityId: Int = 0,
)
